geoh5py/data/float_data.py

```diff
@@ -50,11 +51,17 @@ def values(self, values):
         self._values = self.check_vector_length(values)
 
     def check_vector_length(self, values) -> np.ndarray:
```

Comment:
**suggestion:** somewhat unclear semantic

since this is a member function, called by another function to fixup `self._values`, semantic would be clearer if it simply updates `self._values` instead of returning it.

Also when it returns, there is no guaranty that `self.n_values == len(self._values)`
Should it just be named `flatten_values()`?
Are there any case where we would keep non-flatten values?

Reply:
Just for context, something like this is needed because geoh5 can store vectors shorter than n_vertices if the bottom is all ndv.

I m not fully following you here, let's talk about it.